<!--
  arbe_phoenix_radar_driver video_recorder launch 
 -->
<launch>

    <arg name="video" default="0" />
    <arg name="video_width" default="1280" />
    <arg name="video_height" default="720" />
    <arg name="video_codec" default="mp42" />
    <arg name="video_path" default="arbe_cam_[timestamp].avi" />
    <arg name="screen_capture_path" default="arbe_screen_capture_[timestamp].avi" />
    <arg name="video_topic" default="/video_recorder/image_raw" />
    <arg name="camera_topic" default="/cv_camera/image_raw" />
    <arg name="screen_capture_topic" default="/image" />

    <arg name="update_rate" default="10" />
    <arg name="use_image_show" default="false" />
    <arg name="use_image_view" default="false" />

    <node name="nodelet_manager" pkg="nodelet" type="nodelet"
      args="manager" output="screen" />

    <node name="screen_grab" pkg="nodelet" type="nodelet"
      args="load screen_grab/ScreenGrab nodelet_manager" output="screen" >
        <param name="update_rate" value="$(arg update_rate)" />
        <param name="x_offset" value="0" />
        <param name="y_offset" value="0" />
        <param name="width" value="1920" />
        <param name="height" value="1080" />
    </node>

    <node name="video_recorder" pkg="video_recorder" type="recorder.py" output="screen" required="true">

        <param name="output_width" type="int" value="1920" />
        <param name="output_height" type="int" value="1080" />
        <param name="output_format" value="$(arg video_codec)" />
        <param name="output_path" value="$(arg video_path)" />

        <param name="output_topic" value="$(arg video_topic)" />

        <param name="source1" value="/cv_camera/image_raw,0,0,1920,1080" />
    </node>

    <node name="screen_recorder" pkg="video_recorder" type="recorder.py" output="screen" required="true">

        <param name="output_width" type="int" value="1920" />
        <param name="output_height" type="int" value="1080" />
        <param name="output_format" value="$(arg video_codec)" />
        <param name="output_path" value="$(arg screen_capture_path)" />

        <param name="output_topic" value="$(arg video_topic)" />
        <param name="camera_topic" value="$(arg screen_capture_topic)" />

        <param name="source1" value="/image,0,0,1920,1080" />
    </node>

</launch>

